packages/python/accern_data/accern_data.py

```diff
@@ -530,40 +591,45 @@ def _scroll(
             except pd.errors.EmptyDataError:
                 break
 
-    def _process_date(
+    def _get_valid_mode(
             self,
-            cur_date: str,
-            output_path: str,
-            output_pattern: Optional[str],
-            *,
-            is_first_time: bool,
-            mode: Mode,
+            mode: Optional[Union[Mode, ModeType, Tuple[ModeType, bool]]],
+            deep_copy: bool) -> Mode:
```

Comment:
what is the purpose of this?

Reply:
if we do provide deepcopy of mode to `iterate_range`, then we will not get columns to our data.
mode inside `iterate_range` get's column names. If it is a reference to the mode in `download_range` function then columns will be there while writing them to files.

Reply:
no, iterate_range handles all of this. you just pass the original mode arg to that function. download_range should not worry about that. mode implements how to write. however, iterate_range needs to be able to communicate the mode to download_range. you could for example pass a `Callback[[Mode], None]` that gets called by iterate_range with the correct mode object and use what is passed there as the mode object